enhancements/node-tuning/pao-in-nto.md

```diff
@@ -0,0 +1,317 @@
+
+This should be kept as-is, the Telco team will own all the bugs. We might need to move the component under NTO or rename it.
+
+The bugzilla component is currently not part of OCP metrics and we can include it. TBD
+
+
+### User Stories
+
+Both the NTO and PAO functionality should be kept intact and no regressions are allowed. The user should not notice anything except the fact that OLM is no longer needed to enable the tuning functionality.
+
+### Implementation Details/Notes/Constraints
+
+- Existing api compatibility practices:
+PAO API has been treated as Tier 1 for multiple releases already (since 4.4).
+
+- Enumerate list of performance profile owned objects:
+MachineConfig, KubeletConfiguration, Tuned (see [diagram](https://github.com/openshift-kni/performance-addon-operators/blob/master/docs/interactions/diagram.png)).
+All activity happening on the cluster or nodes is indirect via other components (MCO, NTO) reacting to changes in objects owned by PAO.
+
+- Enumerate list of webhooks:
+Validation and conversion webhook (TBD)
+
+- Updates to openshift conformance test suite if included by default:
+All basic checks are doable on all linux platforms except when real-time kernel is needed. That case used to be only supported on GCP (and bare metal).
+Measuring real latency requires bare metal, but is not needed for conformance.
+
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions
+
+- CI: Integration with NTO is currently TBD.
+
+- CI: One of the test suites (4_latency) implements an opinionated latency testing suite to get comparable results from customers and to prevent configuration / command line invocation mistakes. Keeping it in PAO/NTO or moving it into cnf-tests is TBD.
+
+- Maintainability: There are two options:
+  - Maintaining a separate copy of the code in the NTO repository.
+  - Vendoring the current PAO code as a "library" until we can stop supporting the standalone mode (4.6+).
+  From OCP perspective it makes little difference as it would just be part of NTO release process indeed.
+
+- API definitions: They would need to go into openshift/api but at this moment NTO does include its own API definitions under openshift/api.
+The NTO API definitions transition into openshift/api should be considered to be done at the time of PAO/NTO merge.
+
+- List of pertinent topology options (control plane and infrastructure) where this operator is pertinent:
+The assumption is that PAO is not installed at all when control plane topology is "external" and the infrastructure topology is not pertinent to the function
```

Comment:
If the PAO becomes part of the NTO, do we achieve this by not installing the NTO? Is that already how NTO is handled for external control planes?

Reply:
@jmencak   Can you affirm that this is the way NTO is handled for external control planes ?

Reply:
My understanding is that currently, IBM Cloud is the only cloud with external control plane topology.  NTO *is* being installed on IBM Cloud by using a [deployment manifest (annotation)](https://github.com/openshift/cluster-node-tuning-operator/blob/master/manifests/50-operator-ibm-cloud-managed.yaml#L5-L6).  Unless we merge PAO into NTO in a way that we can make use of the above deployment manifest (not what I believe the plan is), similar manifest for including PAO in IBM Cloud would be necessary.  Therefore, as things are planned now, PAO would *not* run on external control planes.

Currently, it is unclear to me what (if anything) needs to be done to include NTO (and PAO for that matter) on other external control plane topologies wrt. efforts such as HyperShift which will make it possible to run external control planes on platforms other than IBM Cloud.